File: bmi.rs
use std::io::{self, Write};
use std::error::Error;

fn main() -> Result<(), Box<Error>> {
	print!("Enter your weight (kg): ");
	flush()?;
	let weight: f64 = read_line()?.parse()?;

	print!("Enter your height (m): ");
	flush()?;
	let height: f64 = read_line()?.parse()?;
	
	let bmi = weight / height.powf(2.0);
	println!("Your BMI is {:.2}", bmi);

	Ok(())
}

fn read_line() -> io::Result<String> {
	let mut line = String::new();
	io::stdin().read_line(&mut line).map(|_| line.trim().into())
}

fn flush() -> io::Result<()> {
	io::stdout().flush()
}

File: bmi.r
if (interactive()) {

	weight = as.numeric(readline(prompt='Your Weight (kg): '))
	height = as.numeric(readline(prompt='Your Height (m): '))

} else {

	cat('Your Weight (kg): ')
	weight = as.numeric(readLines('stdin', n=1))
	
	cat('Your Height (m): ')
	height = as.numeric(readLines('stdin', n=1))

}

bmi = weight / (height**2)

print(paste('Your bmi is', round(bmi, 2)))
